From 9e9f4052abd95e7fa46a5e7ba0bbd37c338f328c Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:38:15 -0400
Subject: [ethernet] i40e/i40evf: remove rx_errors and rx_missed

Message-id: <1409658042-22689-135-git-send-email-sassmann@redhat.com>
Patchwork-id: 90329
O-Subject: [RHEL7.1 PATCH 134/281] i40e/i40evf: remove rx_errors and rx_missed
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Shannon Nelson <shannon.nelson@intel.com>

The rx_errors (GLV_REPC) and rx_missed (GLV_RMPC) were removed
from the chip design.

Change-ID: Ifdeb69c90feac64ec95c36d3d32c75e3a06de3b7
Signed-off-by: Shannon Nelson <shannon.nelson@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 03da6f6a4ff31987ee988f15840f005f3ebf4f45)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_debugfs.c b/drivers/net/ethernet/intel/i40e/i40e_debugfs.c
index 1aaec40..1bd0adb 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_debugfs.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_debugfs.c
@@ -862,12 +862,11 @@ static void i40e_dbg_dump_eth_stats(struct i40e_pf *pf,
    "    rx_bytes = \t%lld \trx_unicast = \t\t%lld \trx_multicast = \t%lld\n",
   estats->rx_bytes, estats->rx_unicast, estats->rx_multicast);
  dev_info(&pf->pdev->dev,
-   "    rx_broadcast = \t%lld \trx_discards = \t\t%lld \trx_errors = \t%lld\n",
-   estats->rx_broadcast, estats->rx_discards, estats->rx_errors);
+   "    rx_broadcast = \t%lld \trx_discards = \t\t%lld\n",
+   estats->rx_broadcast, estats->rx_discards);
  dev_info(&pf->pdev->dev,
-   "    rx_missed = \t%lld \trx_unknown_protocol = \t%lld \ttx_bytes = \t%lld\n",
-   estats->rx_missed, estats->rx_unknown_protocol,
-   estats->tx_bytes);
+   "    rx_unknown_protocol = \t%lld \ttx_bytes = \t%lld\n",
+   estats->rx_unknown_protocol, estats->tx_bytes);
  dev_info(&pf->pdev->dev,
    "    tx_unicast = \t%lld \ttx_multicast = \t\t%lld \ttx_broadcast = \t%lld\n",
    estats->tx_unicast, estats->tx_multicast, estats->tx_broadcast);
diff --git a/drivers/net/ethernet/intel/i40e/i40e_ethtool.c b/drivers/net/ethernet/intel/i40e/i40e_ethtool.c
index 98037ee..205b559 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_ethtool.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_ethtool.c
@@ -78,7 +78,6 @@ static int i40e_add_fdir_ethtool(struct i40e_vsi *vsi,
 static struct i40e_stats i40e_gstrings_stats[] = {
  I40E_PF_STAT("rx_bytes", stats.eth.rx_bytes),
  I40E_PF_STAT("tx_bytes", stats.eth.tx_bytes),
- I40E_PF_STAT("rx_errors", stats.eth.rx_errors),
  I40E_PF_STAT("tx_errors", stats.eth.tx_errors),
  I40E_PF_STAT("rx_dropped", stats.eth.rx_discards),
  I40E_PF_STAT("tx_dropped", stats.eth.tx_discards),
diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index 3f327ec..93ec7ea 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -803,8 +803,6 @@ static void i40e_update_vsi_stats(struct i40e_vsi *vsi)
 
  /* update netdev stats from eth stats */
  i40e_update_eth_stats(vsi);
- ons->rx_errors = oes->rx_errors;
- ns->rx_errors = es->rx_errors;
  ons->tx_errors = oes->tx_errors;
  ns->tx_errors = es->tx_errors;
  ons->multicast = oes->rx_multicast;
diff --git a/drivers/net/ethernet/intel/i40e/i40e_type.h b/drivers/net/ethernet/intel/i40e/i40e_type.h
index 02e8f9a..5a930b3 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_type.h
+++ b/drivers/net/ethernet/intel/i40e/i40e_type.h
@@ -972,8 +972,6 @@ struct i40e_eth_stats {
  u64 rx_multicast;  /* mprc */
  u64 rx_broadcast;  /* bprc */
  u64 rx_discards;  /* rdpc */
- u64 rx_errors;   /* repc */
- u64 rx_missed;   /* rmpc */
  u64 rx_unknown_protocol; /* rupp */
  u64 tx_bytes;   /* gotc */
  u64 tx_unicast;   /* uptc */
-- 
1.7.1